Makefile

```diff
@@ -353,9 +353,21 @@ lock-package-images: check-carvel # Updates the image lock file for a package. U
 	@printf "\n===> Updating image lockfile for package $${PACKAGE}/$${VERSION}\n";\
 	cd addons/packages/$${PACKAGE}/$${VERSION} && kbld --file bundle --imgpkg-lock-output bundle/.imgpkg/images.yml >> /dev/null;\
 
-push-package: check-carvel # Build and push a package template. Tag will default to `latest`. Usage: make push-package PACKAGE=foobar VERSION=1.0.0
+push-package: check-carvel # Verify openAPIv3 schema in package before build and push a package template. Tag will default to `latest`. Usage: make push-package PACKAGE=foobar VERSION=1.0.0
 	@printf "\n===> pushing $${PACKAGE}/$${VERSION}\n";\
-	cd addons/packages/$${PACKAGE}/$${VERSION} && imgpkg push --bundle $(OCI_REGISTRY)/$${PACKAGE}:$${VERSION} --file bundle/;\
+	./hack/packages/verify-openapischema-for-package.sh $(PACKAGE) $(VERSION) \
+	&& cd addons/packages/$${PACKAGE}/$${VERSION} && imgpkg push --bundle $(OCI_REGISTRY)/$${PACKAGE}:$${VERSION} --file bundle/;\
+
+generate-openapischema-package: #Generate package with OpenAPI v3 schema
```

Comment:
I don't really know much about how the packaging stuff works, so please bear with me. Is this `generate-openapischema-package` specific to a particular package? It doesn't seem like it since there is a `$${PACKAGE}` var out there, but I'm wondering if we should put any kind of package pre or post processing into a sub-Makefile located in the `/addons` or `/addons/packages` folder as not to bloat the upper level Makefile. Thoughts?

cc: @seemiller @jpmcb

Reply:
When I was adding make target, I was recommended by @seemiller to add it to the main Makefile and yes, it is used to generate openapischema-package specific to a particular package.
Also, make push package which is now modified will only look and verify  generated schema for packages which have schema.yaml file

Reply:
my comment here is just more like just an open question to people who are better suited/qualified to answer. just wanted to bring awareness to try not bloat the main Makefile if possible. if this is the best or only place for this, then totally cool by me. adding @seemiller

Reply:
@dvonthenen I like your idea of moving these to a more package specific makefile. However, could we do that as it's own separate issue?